### ISSUE TITLE
Problem with Confluent.Kafka version 2.1.0

### ISSUE THREAD

**Issue by janek9971** [kmcclellan/kafka-hosting#11]
Hello,

I have a problem with

` services.AddKafkaClient().Configure(opts =>
            {
                opts.Configure(options);
            });`
            
when using Confluent.Kafka v.2.1.0 from April 6, I get an error.

Exception message:

> Method 'SetSaslCredentials' in type 'Confluent.Kafka.DependencyInjection.ScopedProducer`2' from assembly 'Confluent.Kafka.DependencyInjection, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.

With previous v.2.0.2 from January 23 everything is working great

**Comment by janek9971** [comment]
Ok I found the problem by downloading the repo. All interfaces which derive from IClient, must now implement method SetSaslCredentials

![image](https://user-images.githubusercontent.com/10576321/232287114-923bd89c-0dba-44c7-ac80-79fba94e57e8.png)

If I understand the code correctly this should fix the problem right?

```
sealed class ScopedConsumer<TKey, TValue> : IConsumer<TKey, TValue>
{
    readonly IDisposable scope;
    readonly IConsumer<TKey, TValue> inner;

    bool closed;

    public ScopedConsumer(ClientScopeFactory initOptions)
    {
        this.scope = initOptions(out var options);
        this.inner = options.CreateConsumer<TKey, TValue>();
    }

    public void SetSaslCredentials(string username, string password)
    {
        this.inner.SetSaslCredentials(username, password);
    }

    public TopicPartitionOffset PositionTopicPartitionOffset(TopicPartition partition)
    {
        return this.inner.PositionTopicPartitionOffset(partition);
    }
...
}
```

**Comment by kmcclellan** [comment]
Yup thanks for the heads up. Never know when the maintainers are gonna release breaking changes 😅. I just released a fix for this if you update Confluent.Kafka.DependencyInjection to v3.0.1.

**Comment by kmcclellan** [comment]
Fixed by f63bdc2fed1191c5518066fc5e8f6b431d3c7079

**Comment by janek9971** [comment]
Thanks :)
